cmake/DetectOptions.cmake

```diff
@@ -131,6 +131,17 @@ if(NOT (CMAKE_CXX_COMPILER_ID STREQUAL "PGI") AND NOT MSVC)
     endif()
 endif()
 
+# Table
```

Comment:
So, the enable logic here is wrong and doesn't really handle the `ON` case correctly, but I see it's the same logic used in several other places so it's fine to leave it here for this PR.  I'll look at addressing all of them in a separate PR after this.

Reply:
@chuckatkins Thanks and please go ahead after I merge. That would be very helpful. I would also like to take a step further and see if there is a way to make the nlohmann::json library work with the PGI compiler and MSVC. So far I have four engines using JSON and we will probably use JSON in the query interface as well. So if this breaks the windows build then it's going to be a blocking issue eventually. I am also looking into https://github.com/Tencent/rapidjson/ and see if we can replace the JSON library with this one, which is said to be faster than the nlohmann implementation. Your comments or ideas would be very helpful.